src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,151 +201,160 @@ public boolean canBeRotated() {
-            if (machineActive && !pulverizerPaused) {
+        if (this.canWork()) {
+            if (machineTier == MachineTier.TIER_0) {
+                fuelRemaining--;
+                machineActive = fuelRemaining > 0;
+            } else if (ticksRemaining > 0) {
+                container.takePower(powerUsage, false);
+                machineActive = true;
+            }
+            if (machineActive && ticksRemaining > 0) {
                 ticksRemaining--;
-
-                if (machineTier != MachineTier.TIER_0)
-                    this.container.takePower(powerUsage, false);
-                if (fuelRemaining > 0)
-                    fuelRemaining--;
-
             }
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-            if (inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
-                ItemStack itemIn = inventory.getStackInSlot(0);
-                ItemStack itemOut;
-
-                if (!PulverizerRegistry.containsInput(itemIn))
-                    return;
-
-                if (itemIn.stackSize - 1 <= 0) {
-                    itemOut = itemIn.copy();
-                    itemIn = null;
-                } else {
-                    itemOut = itemIn.copy();
-
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
-
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
-
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
-
-                ticksRemaining = 200;
-                machineActive = true;
-
-                this.markForUpdate();
-                this.markDirty();
+        if (fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0) {
+            if (inventory.getStackInSlot(1) != null) {
+                burnTime(1);
+            }
+            if (inventory.getStackInSlot(0) != null) {
+                burnTime(0);
             }
+        }
+
 
+        if (this.canWork() && inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
+            if (!cancrush(inventory.getStackInSlot(0)))
+                return;
 
-            if (fuelRemaining == 0 && machineActive && machineTier == machineTier.TIER_0) {
-                ticksRemaining = 0;
-                test = false;
+            ItemStack itemIn = inventory.getStackInSlot(0);
+            ItemStack itemOut;
 
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
             } else {
-                test = false;
+                itemOut = itemIn.copy();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS134)